cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -65,14 +65,14 @@
   private final ExploreFacade exploreFacade;
   private final boolean allowDatasetUncheckedUpgrade;
   private final UsageRegistry usageRegistry;
-  private final AbstractNamespaceClient namespaceClient;
+  private final NamespaceStore nsStore;
```

Comment:
Agreed. But due to the current limitation in NamespaceAdmin its not possible within the release/3.3.1 scope. So, I have opened a jira here: https://issues.cask.co/browse/CDAP-4904